web/modules/custom/os2forms_email/src/Plugin/WebformHandler/OS2FormsEmailWebformHandler.php

```diff
@@ -0,0 +1,257 @@
+
+    $webform = $webform_submission->getWebform();
+    $settings = $webform->getThirdPartySetting('os2forms', 'os2forms_email');
+
+    if ($settings['enabled'] && !empty($settings['emails'])) {
+      $this->handleAttachmentNotification($webform_submission, $message, $settings['emails']);
+    }
+
+    return parent::sendMessage($webform_submission, $message);
+  }
+
+  /**
+   * Handles attachment notification on submission.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webform_submission
+   *   A webform submission.
+   * @param array $message
+   *   An array of message parameters.
+   * @param string $emails
+   *   A string of emails.
+   */
+  private function handleAttachmentNotification(WebformSubmissionInterface $webform_submission, array $message, string $emails): void {
+    if ($this->isAttachmentFileSizeThresholdSurpassed($webform_submission)) {
+      $this->sendFileSizeNotification($webform_submission, $message, $emails);
+    }
+  }
+
+  /**
+   * Checks whether file size threshold is surpassed by submission.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webform_submission
+   *   A webform submission.
+   *
+   * @return bool
+   *   Whether threshold is surpassed or not.
+   */
+  private function isAttachmentFileSizeThresholdSurpassed(WebformSubmissionInterface $webform_submission): bool {
+    // Determine file size threshold in bytes.
+    $threshold = $this->configFactory->get('os2forms_email')->get('notification_file_size_threshold') ?? self::DEFAULT_ATTACHMENT_FILE_SIZE_THRESHOLD;
+    $thresholdInBytes = $this->convertToBytes($threshold);
+
+    $fileElementIds = $this->getRelevantFileIdsFromSubmission($webform_submission);
+
+    $totalSize = 0;
+    foreach ($fileElementIds as $fileElementId) {
+      $fileElement = File::load($fileElementId);
+      $totalSize += (int) $fileElement->getSize();
+    }
+
+    return $totalSize > $thresholdInBytes;
+  }
+
+  /**
+   * Gets array of file ids in submission that are attached in email.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $submission
+   *   A webform submission.
+   *
+   * @return array
+   *   File ids.
+   */
+  private function getRelevantFileIdsFromSubmission(WebformSubmissionInterface $submission): array {
+    $elements = $submission->getWebform()->getElementsDecodedAndFlattened();
+
+    // Removed excluded elements.
+    $excludedElements = $this->configuration['excluded_elements'];
+
+    foreach ($excludedElements as $key => $excludedElement) {
+      if (array_key_exists($key, $elements)) {
+        unset($elements[$key]);
+      }
+    }
+
+    $fileElements = [];
+
+    foreach (self::FILE_ELEMENT_TYPES as $fileElementType) {
+      $fileElements[] = $this->getAvailableElementsByType($fileElementType, $elements);
+    }
+
+    // https://dev.to/klnjmm/never-use-arraymerge-in-a-for-loop-in-php-5go1
+    $fileElements = array_merge(...$fileElements);
```

Comment:
Array functions are your friend (you've already used them at least twice in this pull request):

```suggestion
    $fileElements = array_map(
      fn ($type) => $this->getAvailableElementsByType($type, $elements),
      self::FILE_ELEMENT_TYPES
    );

    // Flatten the array.
    // https://dev.to/klnjmm/never-use-arraymerge-in-a-for-loop-in-php-5go1
    $fileElements = array_merge(...$fileElements);
```